cli/audio_codec.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+extern "C"
+{
+#include <libavformat/avformat.h>
+#include <libavutil/time.h>
+#include <libavutil/file.h>
+#include <libavcodec/avcodec.h>
+#include <libswresample/swresample.h>
+
+#include <libavutil/opt.h>
+#include <libavutil/channel_layout.h>
+#include <libavutil/samplefmt.h>
+}
+
+#define INPUT_BUFFER_SIZE   (8<<20)
+#define STREAM_READ_SIZE    (512<<10)           // has to be larger than 128KB
```

Comment:
nit:
constexpr const uint64_t

Let's also refrain from putting things in global scope.  namespaces / structs / typedef enums etc.